internal/modelcontroller/model_source.go

```diff
@@ -214,28 +215,46 @@ func (r *ModelReconciler) pvcPodAdditions(url modelURL) *modelSourcePodAdditions
 	}
 }
 
-var modelURLRegex = regexp.MustCompile(`^([a-z]+):\/\/(\S+)$`)
+var modelURLRegex = regexp.MustCompile(`^([a-z]+):\/\/([^?]+)(\?.*)?$`)
 
 func parseModelURL(urlStr string) (modelURL, error) {
 	matches := modelURLRegex.FindStringSubmatch(urlStr)
-	if len(matches) != 3 {
+	if len(matches) != 3 && len(matches) != 4 {
 		return modelURL{}, fmt.Errorf("invalid model URL: %s", urlStr)
 	}
 	scheme, ref := matches[1], matches[2]
 	name, path, _ := strings.Cut(ref, "/")
+	modelParam := ""
+
+	// check to see if the pvc model has a query string, set model to query string, path to host
+
+	if scheme == "pvc" {
```

Comment:
I would remove the `scheme == "pvc"` check. I dont think it is needed, and it might cause confusion as to why the model param is not getting parsed in the future if we support `scheme == "pv"` or something else.